### PR TITLE
Clientset for seldon operator

### PR DESCRIPTION
**Pull request by SachinVarghese**
This PR does the following 
- the seldon operator is migrated to a multi-group structure.
- added a make script to create clientsets for the multi-version operator `make create-client`
- operator client sets for all the existing versions of seldonDeployment CRD

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **sachinvarghese**

If they are not already assigned, you can assign the PR to them by writing `/assign @sachinvarghese` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Fri Dec 13 12:27:18 UTC 2019
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1258/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1258 --build=1

**Comment by seldondev** [comment]
Fri Dec 13 12:27:20 UTC 2019
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1258/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1258 --build=2

File: operator/Makefile [hunk 1]
```diff
@@ -103,6 +103,10 @@ vet:
 generate: controller-gen
 	$(CONTROLLER_GEN) object:headerFile=./hack/boilerplate.go.txt paths="./..."
 
+# Generate Clientset
+create-client: test
+	./hack/update-codegen.sh
```
Comment (adriangonz) [inline operator/Makefile]: Doesn't `update-codegen.sh` just download the generator tools?

Comment (SachinVarghese) [reply operator/Makefile]: And create the client folder.

Comment (adriangonz) [reply operator/Makefile]: Got it! Thanks for clarifying that.

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/SeldonIO/seldon-core/pull/1258#issuecomment-565453083" title="Approved">cliveseldon</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [cliveseldon]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Fri Dec 13 14:08:45 UTC 2019
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1258/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1258 --build=3

**Comment by seldondev** [comment]
Fri Dec 13 14:08:49 UTC 2019
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1258/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1258 --build=4

**Pull request merged**